src/app/clusters/door-lock-server/door-lock-server-user.cpp

```diff
@@ -725,12 +730,12 @@ bool emberAfDoorLockClusterUnlockDoorCallback(chip::app::Command * commandObj, u
                                (uint8_t *) &rfOperationEventMask, sizeof(rfOperationEventMask));
 
     // send operation event
-    if (doorUnlocked && (rfOperationEventMask & EMBER_BIT(2)) && (pin != NULL))
+    if (doorUnlocked && (rfOperationEventMask & EMBER_BIT(2)) && (pin.size() != 0))
```

Comment:
```suggestion
    if (doorUnlocked && (rfOperationEventMask & EMBER_BIT(2)) && !pin.empty())
```
And same in a bunch of other places.